src/IronPythonExtension/IronPythonExtension.cs

```diff
@@ -49,16 +49,26 @@ internal void OnMessageLogged(ILogMessage msg)
         /// </summary>
         /// <param name="sp"></param>
         public void Startup(StartupParams sp)
+        {
+            // Do nothing for now
+        }
+
+        /// <summary>
+        /// Action to be invoked when the Dynamo has started up and is ready
+        /// for user interaction. 
+        /// </summary>
+        /// <param name="sp"></param>
+        public void Ready(ReadyParams sp)
         {
             // Searches for DSIronPython engine binary in same folder with extension itself
             var targetDir = Path.GetDirectoryName(Assembly.GetAssembly(typeof(IronPythonExtension)).Location);
-            var libraryLoader = sp.LibraryLoader;
+            var libraryLoader = sp.StartupParams.LibraryLoader;
```

Comment:
How did this work before? Is LibraryLoader a member of both ReadyParams and StartupParams?

Reply:
`LibraryLoader` is only available in `Startupparams`. Previously we are only loading the IronPython engine when Dynamo launches -> loading extension -> load Ironpython engine. Switching to the place here, it will work for the additional installing the extension in the session case as I illustrated in gif

Reply:
I understood that but how did the code compile before? I don't see a LibraryLoader property on ReadyParams?

Reply:
Maybe I'm missing something obvious.

Reply:
@aparajit-pratap Previously we have `StartupParams` available but now we need to access it from `ReadyParams`

Reply:
Ah, sorry, I knew I was missing something. I'm going blind.

Reply:
@aparajit-pratap No worries. I blame our interface 😉